### PR TITLE
fix(object-alt): ignore unloaded objects

### PR DESCRIPTION
**Pull request by WilcoFiers**
Skip objects on the object-alt rule if they either have no data attribute, or if any of their fallback content is visible.

Closes issue: #3656

I was having difficulty getting the ACT tests to pass because the server doesn't correctly map the URLs. I opened a separate PR for that: #3681

### CODE REVIEW

File: test/integration/rules/object-alt/object-alt.html [hunk 1]
```diff
@@ -1,18 +1,75 @@
...
+<object id="violation1" data="data:text/html,Object%20content"></object>
+<object id="violation2" data="data:text/html,Object%20content">
+  <div></div>
+</object>
+<object id="violation3" data="data:text/html,Object%20content">
+  This object has text.
+</object>
+<object
+  id="violation4"
+  role="none"
```
Comment (straker) [inline test/integration/rules/object-alt/object-alt.html:35]: This fails [inapplicable example 5](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/8fc3b6/proposed/#inapplicable-example-5)

File: test/integration/rules/object-alt/object-alt.html [hunk 2]
```diff
@@ -1,18 +1,75 @@
...
+  tabindex="0"
+  data="data:text/html,Object%20content"
+></object>
+<object
+  id="violation5"
+  role="presentation"
```
Comment (straker) [inline test/integration/rules/object-alt/object-alt.html:41]: This fails [inapplicable example 5](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/8fc3b6/proposed/#inapplicable-example-5)

File: test/integration/rules/object-alt/object-alt.html [hunk 3]
```diff
@@ -1,18 +1,75 @@
...
+></object>
+<object
+  id="violation7"
+  role="presentation"
+  aria-live="assertive"
+  data="data:text/html,Object%20content"
+></object>
+<object
+  id="violation8"
+  role="img"
```
Comment (straker) [inline test/integration/rules/object-alt/object-alt.html:59]: This fails [inapplicable example 1](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/8fc3b6/proposed/#inapplicable-example-1)

File: test/integration/rules/object-alt/object-alt.html [hunk 4]
```diff
@@ -1,18 +1,75 @@
...
-<object id="violation3">This object has text.</object>
-<object id="violation4" role="none" tabindex="0"></object>
-<object id="violation5" role="presentation" tabindex="0"></object>
-<object id="violation6" role="none" aria-live="assertive"></object>
-<object id="violation7" role="presentation" aria-live="assertive"></object>
+<object id="violation1" data="data:text/html,Object%20content"></object>
+<object id="violation2" data="data:text/html,Object%20content">
+  <div></div>
+</object>
+<object id="violation3" data="data:text/html,Object%20content">
```
Comment (straker) [inline test/integration/rules/object-alt/object-alt.html:30]: This fails [inapplicable example 6](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/8fc3b6/proposed/#inapplicable-example-6)

**Review by straker — DISMISSED** [review]
I don't thing this is the right approach. Before we only failed 1 of the inapplicable examples (the one this pr fixed), but with the change we now fail 3 others. From reading the ACT rule I don't think it requires looking at if a resource loaded or not. The applicability states that it should only be run on object elements that match the following criteria:

> * The object element is included in the accessibility tree; and
> * The object element has no explicit role; and
> * The object element embeds a resource whose MIME type is either image, or audio or video.

If we want to be more conformant, should we look at the 3rd point rather than if content is loaded or not?

File: test/integration/rules/object-alt/object-alt.html (hunk 3 above)
Comment (WilcoFiers) [reply test/integration/rules/object-alt/object-alt.html:59]: ACT Covers this in a different rule. We don't.

File: test/integration/rules/object-alt/object-alt.html (hunk 4 above)
Comment (WilcoFiers) [reply test/integration/rules/object-alt/object-alt.html:30]: ACT doesn't cover this, but this is still an issue.

**Review by straker — APPROVED** [review]

**Pull request merged**